### PR TITLE
Allow to use regex on the options skip_dir and skip_file #123

### PR DESCRIPTION
**Pull request by JonaMX**
I was trying to sync a single directory and a single file extension #123, but I realized that the pattern was being modified by the function [wild2Regex](https://github.com/skilion/onedrive/blob/master/src/util.d#L53-L79), that's why I added the option to use regex and skip this function.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,6 +41,8 @@ Available options:
 * `sync_dir`: directory where the files will be synced
 * `skip_file`: any files that match this pattern will be skipped during sync
 * `skip_dir`: any directories that match this pattern will be skipped during sync
+* `use_regex`: use regex for `skip_dir` and `skip_file` options, default `false`
```
Comment (thomasfedb) [inline README.md]: I would change `regex` to `regular expressions`

File: README.md [hunk 2]
```diff
@@ -41,6 +41,8 @@ Available options:
...
+* `ignore_local_changes`: change to `yes` if you want to ignore local changes, default `no`
```
Comment (thomasfedb) [inline README.md]: Here you're using `'yes'` and `'no'`, but above you're using `'true'` and `'false'`. Why?

Comment (thomasfedb) [inline README.md]: Also, it seems that this should be a separate PR to the regex changes.

File: src/monitor.d [hunk 3]
```diff
@@ -41,8 +41,14 @@ struct Monitor
...
 		this.verbose = verbose;
-		skipDir = wild2regex(cfg.getValue("skip_dir"));
-		skipFile = wild2regex(cfg.getValue("skip_file"));
+                // Check if we should parse the regex or not
+                if (cfg.getValue("use_regex") == "true") {
+                        skipDir = regex(cfg.getValue("skip_dir"));
+                        skipFile = regex(cfg.getValue("skip_file"));
+                } else {
+                        skipDir = wild2regex(cfg.getValue("skip_dir"));
+                        skipFile = wild2regex(cfg.getValue("skip_file"));
```
Comment (thomasfedb) [inline src/monitor.d]: This changed should be moved into a function that handles choosing between wildcard and regex, this code is very WET.

File: src/sync.d [hunk 4]
```diff
@@ -64,8 +64,14 @@ final class SyncEngine
...
 		this.itemdb = itemdb;
-		skipDir = wild2regex(cfg.getValue("skip_dir"));
-		skipFile = wild2regex(cfg.getValue("skip_file"));
+                // Check if we should parse the regex or not
+                if (cfg.getValue("use_regex") == "true") {
+                        skipDir = regex(cfg.getValue("skip_dir"));
+                        skipFile = regex(cfg.getValue("skip_file"));
+                } else {
+                        skipDir = wild2regex(cfg.getValue("skip_dir"));
+                        skipFile = wild2regex(cfg.getValue("skip_file"));
```
Comment (thomasfedb) [inline src/sync.d]: And the same here. Ideally parsing of `skip_dir` and `skip_file` could be moved to a single location.

**Comment by JonaMX** [comment]
Thanks for the feedback @thomasfedb, you are right, I'll create two different PRs, one for the regular expressions, and the other one for ignoring the local changes.

**Comment by skilion** [comment]
Not needed anymore with selective sync
